cachelib/navy/common/tests/DeviceTest.cpp

```diff
@@ -196,6 +198,70 @@ TEST(Device, Stats) {
   device.getCounters({toCallback(visitor)});
 }
 
+// Test for FdpNvme Constructor
+TEST(FDP, InitializationTest) {
+  int nsId = 1;
+  uint32_t lbaShift = 12;
+  uint32_t maxTfrSize = 262144;
+  uint64_t startLba = 0;
+
+  NvmeData data(nsId, lbaShift, maxTfrSize, startLba);
+  struct nvme_fdp_ruh_status ruh_status{};
+  ruh_status.nruhsd = 10;
+
+  EXPECT_NO_THROW(FdpNvme fdp = FdpNvme(data, &ruh_status));
+}
+
+// Test the Fdp Handle allocation
+TEST(FDP, FdpHandleAllocationTest) {
+  int nsId = 1;
+  uint32_t lbaShift = 12;
+  uint32_t maxTfrSize = 262144;
+  uint64_t startLba = 0;
+
+  NvmeData data(nsId, lbaShift, maxTfrSize, startLba);
+  struct nvme_fdp_ruh_status ruh_status{};
+  ruh_status.nruhsd = 4;
+
+  FdpNvme fdp = FdpNvme(data, &ruh_status);
```

Comment:
This is still crashing on me.
`ruh_status` has a field `ruhss`, which is an array of `nvme_fdp_ruh_status_desc` that's not initialized.

In `initializeFDPHandles`, we'll iterate through it as if it has `nruhsd` elements. Maybe initialize it here?

Reply:
@haowu14 We made some changes and made sure that there are no unitialized buffers w.r.t. ruh_status and ruhss. Hopefully this should solve the issue. Pls check it. 
(We could not reproduce the crash in our systems though)